src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -374,260 +363,207 @@ public SymbolicMatch FindMatch(bool isMatch, ReadOnlySpan<char> input, int start
+            int endPos = -1;
+            try
             {
-                if (TStateHandler.IsNullable(ref state, GetCharKind(input, pos + 1)))
-                {
-                    // If the new state accepts the empty string, we found an ending state. Record the position.
-                    endingIndex = pos;
-                }
-                else if (TStateHandler.IsDeadend(ref state))
+                // Loop through each character in the input, transitioning from state to state for each.
+                while (true)
                 {
-                    // If the new state is a dead end, the match ended the last time endingIndex was updated.
-                    currentState = state;
-                    i = pos;
-                    return true;
-                }
-
-                // We successfully transitioned to the next state and consumed the current character,
-                // so move along to the next.
-                pos++;
-            }
-
-            // We either ran out of input, in which case we successfully recorded an ending index,
-            // or we failed to transition to the next state due to the graph becoming too large.
-            currentState = state;
-            i = pos;
-            return false;
-        }
-
-        /// <summary>Find match end position using the original pattern, end position is known to exist. This version also produces captures.</summary>
-        /// <param name="input">input span</param>
-        /// <param name="i">inclusive start position</param>
-        /// <param name="resultRegisters">out parameter for the final register values, which indicate capture starts and ends</param>
-        /// <param name="perThreadData">Per thread data reused between calls.</param>
-        /// <returns>the match end position</returns>
-        private int FindEndPositionCapturing(ReadOnlySpan<char> input, int i, out Registers resultRegisters, PerThreadData perThreadData)
-        {
-            int i_end = input.Length;
-            Registers endRegisters = default;
-            DfaMatchingState<TSet>? endState = null;
+                    // If the state is nullable for the next character, meaning it accepts the empty string,
+                    // we found a potential end state.
+                    if (TStateHandler.IsNullable(ref state, GetCharKind(input, pos)))
+                    {
+                        // Check whether there's a fixed-length marker for the current state.  If there is, we can
+                        // use that length to optimize subsequent matching phases.
+                        matchLength = TStateHandler.FixedLength(ref state);
+                        endPos = pos;
+                        // If this is an isMatch call we are done, since a match is now known to exist.
+                        if (isMatch)
+                            return 1;
```

Comment:
Nit:
```suggestion

                        // If this is an isMatch call we are done, since a match is now known to exist.
                        if (isMatch)
                        {
                            return 1;
                        }
```